frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2787,6 +2787,44 @@ def ones(self, resulting_type: ConversionType) -> Conversion:
 
         return self.add(resulting_type, one, self.zeros(resulting_type))
 
+    def relu(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
+        if x.is_unsigned:
+            if resulting_type.bit_width == x.bit_width:
+                return x
+
+            assert resulting_type.bit_width > x.bit_width
+            return self.extract_bits(resulting_type, x, bits=slice(0, x.original_bit_width))
+
+        if x.original_bit_width == 1:
+            return self.zeros(resulting_type)
+
+        chunk_size = 2
+        intermediate_type = self.tensor(self.eint(chunk_size + 1), shape=x.shape)
+        sign = self.extract_bits(intermediate_type, x, bits=(x.original_bit_width - 1))
+
+        filtered_chunks = []
+        for chunk_start in range(0, x.original_bit_width - 1, chunk_size):
+            chunk_end = min(chunk_start + chunk_size, x.original_bit_width - 1)
+
+            chunk = self.extract_bits(intermediate_type, x, bits=slice(chunk_start, chunk_end))
+
+            shifter = self.constant(self.i(chunk_size + 2), 2)
+            shifted_chunk = self.mul(intermediate_type, chunk, shifter)
+
+            packed_chunk_and_sign = self.add(intermediate_type, shifted_chunk, sign)
+            filtered_chunk = self.tlu(
+                resulting_type,
+                packed_chunk_and_sign,
+                [
+                    (x >> 1) << chunk_start if x % 2 == 0 else 0
```

Comment:
with the sign first, it's itself a ReLU + shift, which is more elegant
why not using bi_variate here ?